tsinfer/formats.py

```diff
@@ -2448,6 +2451,19 @@ def variants(self, sites=None, recode_ancestral=None):
                 geno_map[aa] = 0
                 geno_map[0:aa] += 1
                 genos = geno_map[genos]
+            # Filter out empty alleles, as sgkit pads with them so that all sites have
+            # the same number of alleles. This is only safe if the empty
+            # alleles are at the end of the list, so check this.
+            non_empty_alleles = []
+            empty_seen = False
+            for allele in alleles:
+                if allele != b"" and allele != "":
```

Comment:
How about ``len(allele) == 0``? Above also - seems more robust?

Reply:
Alleles can be `None` though, right? For missing data?

Reply:
`hmm, that's a good point. We stuck `None` as the last allele to stop people using the `-1` genotype as an index. Perhaps there's a better way to do this? It would be nice to allow (if possible) `""` to be used to indicate a deletion.

Note that this is all on the "presenting to tsinfer" side of things. If SGkit allows `None` in the alleles list, it would, I guess, be possible for SGkit to return `None` for an "empty" allele, and then, in the tsinfer format parsing, to remove those `None` values and tag another onto the end if there is missing data. Seems a bit convoluted though.

Reply:
Right... Fair enough.